packages/adp-tooling/src/writer/index.ts

```diff
@@ -42,26 +59,41 @@ export async function generate(basePath: string, config: AdpWriterConfig, fs?: E
     if (!fs) {
         fs = create(createStorage());
     }
-    const tmplPath = join(__dirname, '../../templates/project');
     const fullConfig = setDefaults(config);
 
     fs.copyTpl(join(tmplPath, '**/*.*'), join(basePath), fullConfig, undefined, {
         globOptions: { dot: true },
         processDestinationPath: (filePath: string) => filePath.replace(/gitignore.tmpl/g, '.gitignore')
     });
 
-    // ui5.yaml
-    const ui5ConfigPath = join(basePath, 'ui5.yaml');
-    const baseUi5ConfigContent = fs.read(ui5ConfigPath);
-    const ui5Config = await UI5Config.newInstance(baseUi5ConfigContent);
-    enhanceUI5Yaml(ui5Config, fullConfig);
-    fs.write(ui5ConfigPath, ui5Config.toString());
-    // ui5-deploy.yaml
-    if (hasDeployConfig(fullConfig)) {
-        const ui5DeployConfig = await UI5Config.newInstance(baseUi5ConfigContent);
-        enhanceUI5DeployYaml(ui5DeployConfig, fullConfig);
-        fs.write(join(basePath, 'ui5-deploy.yaml'), ui5DeployConfig.toString());
+    await writeUi5Yaml(basePath, fullConfig, fs);
+
+    return fs;
+}
+
+/**
+ * Writes the adp-project template to the mem-fs-editor instance during migration.
+ *
+ * @param basePath - the base path
+ * @param config - the writer configuration
+ * @param fs - the memfs editor instance
+ * @returns the updated memfs editor instance
+ */
+
+export async function migrateAdp(basePath: string, config: AdpWriterConfig, fs?: Editor): Promise<Editor> {
+    if (!fs) {
+        fs = create(createStorage());
     }
 
+    const fullConfig = setDefaults(config);
+
+    // copy all files except app discriptor
+    fs.copyTpl(join(tmplPath, '**/*.*'), join(basePath), fullConfig, undefined, {
+        globOptions: { dot: true, ignore: ['manifest.appdescr_variant'] },
```

Comment:
@tobiasqueck 
Thank you for the review. I need to exclude the `manifest.appdescr_variant` file, we don't want to call `generate` as is, as `generate` function will create that file. We are creating a `migrate` function to handle migration logic and only write the files that are needed.
The current implementation is to not handle existing ui5.yaml and package.json files, but to generate the new ones. We had a discussion on this with @devinea and agreed we can do it later if needed.

Reply:
aah... there is the difference :) I missed that when looking at the code. Then it is fine